File: DB/Entities/Book.cs
using DB.Enums;
using System;

namespace DB.Entities
{
    public class Book
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public BookType Type { get; set; }
        public DateTime? Year { get; set; }
        public int? PagesAmount { get; set; }

    }
}

File: Web/ClientApp/src/app/Models/book.ts
export class Book {
    id: number;
    title: string;
    author: string;
    type: BookType;
    year: Date;
    pagesAmount: number;
}

export enum BookType {
    Novel = 0,
    Historical = 1,
    Fiction = 2,
    Poetry = 3
}
